crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -423,6 +428,14 @@ where
         to_block: u64,
     ) -> Result<Vec<Log>, FilterError> {
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
+        let is_multi_block_range = from_block != to_block;
+
+        if is_multi_block_range
+            && self.max_blocks_per_filter != 0
```

Comment:
if I read this correctly, then max_blocks_per_filter == 0 will essentially be the same as `max_blocks_per_filter == u64::MAX`, or in other words it will disable limits entirely?

I don't think this is great, this should be solved with an enum type instead:
something like:
```rust
enum BlockRangeLimit {
  None,
  Max(u64)
} 
```

which is essentially an `Option`

Reply:
@mattsse I haven't really understood how type works on rust but seems like it could automatically match the type with None or should we match the type again?

Reply:
this is just intended as a wrapper type to get rid of the `!= 0` check and do `self.max_blocks_per_filter.is_none()` for example, this request is purely cosmetic